src/libraries/ArrayUtil.sol

```diff
@@ -0,0 +1,565 @@
+     */
+
+    // function toInt8(int256[] memory x) internal pure returns (uint8[] memory y) {
+    //     y = new uint8[](x.length);
+    //     for (uint256 i = 0; i < x.length; i++) {
+    //         y[i] = x[i].toUint256().toUint8();
+    //     }
+    // }
+
+    // function toUint8(uint256[] memory x) internal pure returns (uint8[] memory y) {
+    //     y = new uint8[](x.length);
+    //     for (uint256 i = 0; i < x.length; i++) {
+    //         y[i] = x[i].toUint8();
+    //     }
+    // }
+
+    // function toUint64(uint256[] memory x) internal pure returns (uint64[] memory y) {
+    //     y = new uint64[](x.length);
+    //     for (uint256 i = 0; i < x.length; i++) {
+    //         y[i] = x[i].toUint64();
+    //     }
+    // }
+
+    // function toUint64(int256[] memory x) internal pure returns (uint64[] memory y) {
+    //     y = new uint64[](x.length);
+    //     for (uint256 i = 0; i < x.length; i++) {
+    //         y[i] = x[i].toUint256().toUint64();
+    //     }
+    // }
+
+    // function toInt80(uint80[] memory x) internal pure returns (int80[] memory y) {
+    //     y = new int80[](x.length);
+    //     for (uint256 i = 0; i < x.length; i++) {
+    //         y[i] = int80(x[i]);
+    //     }
+    // }
+
+    // function toUint80(int80[] memory x) internal pure returns (uint80[] memory y) {
+    //     y = new uint80[](x.length);
+    //     for (uint256 i = 0; i < x.length; i++) {
+    //         y[i] = uint80(x[i]);
+    //     }
+    // }
+
+    // function toUint80(uint256[] memory x) internal pure returns (uint80[] memory y) {
+    //     y = new uint80[](x.length);
+    //     for (uint256 i = 0; i < x.length; i++) {
+    //         y[i] = x[i].toUint80();
+    //     }
+    // }
+
+    // function toUint80(int256[] memory x) internal pure returns (uint80[] memory y) {
+    //     y = new uint80[](x.length);
+    //     for (uint256 i = 0; i < x.length; i++) {
+    //         y[i] = x[i].toUint256().toUint80();
+    //     }
+    // }
+
+    // function toInt256(uint8[] memory x) internal pure returns (int256[] memory y) {
+    //     y = new int256[](x.length);
+    //     for (uint256 i = 0; i < x.length; i++) {
+    //         y[i] = int256(int8(x[i]));
+    //     }
+    // }
+
+    // function toInt256(uint64[] memory x) internal pure returns (int256[] memory y) {
+    //     y = new int256[](x.length);
+    //     for (uint256 i = 0; i < x.length; i++) {
+    //         y[i] = int256(int64(x[i]));
+    //     }
+    // }
+
+    // function toInt256(uint80[] memory x) internal pure returns (int256[] memory y) {
+    //     y = new int256[](x.length);
+    //     for (uint256 i = 0; i < x.length; i++) {
+    //         y[i] = int256(int80(x[i]));
+    //     }
+    // }
+
+    function toInt256(uint256[] memory x) internal pure returns (int256[] memory y) {
+        y = new int256[](x.length);
+        for (uint256 i = 0; i < x.length; i++) {
+            y[i] = x[i].toInt256();
+        }
+    }
```

Comment:
We should avoid using duplicated naming as the casting function

Reply:
duplicated naming?

Reply:
this is the same as safeCast library. So`array.toInt256()` and `uint.toInt256()` are quite confusing